### PR TITLE
mpv: fix lua library paths and clean up

### PR DESCRIPTION
**Pull request by rnhmjoj**
###### Motivation for this change
I fixed the library paths so luasocket can be used from mpv scripts and removed the libcaca TODO as it seems to work fine. Also I cleaned up a bit the features/options of the package.

###### Things done
- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s)
- [x] Tested compilation of all pkgs that depend on this change using (none)
- [x] Tested execution of all binary files
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by Mic92** [comment]
@GrahamcOfBorg build mpv

File: pkgs/applications/video/mpv/default.nix [hunk 1]
```diff
@@ -13,25 +13,25 @@
     wayland      ? null,
     libxkbcommon ? null
 
-, rubberbandSupport  ? !stdenv.isDarwin, rubberband ? null
+, rubberbandSupport  ? true,  rubberband    ? null
 , xineramaSupport    ? true,  libXinerama   ? null
 , xvSupport          ? true,  libXv         ? null
 , sdl2Support        ? true,  SDL2          ? null
-, alsaSupport        ? !stdenv.isDarwin,  alsaLib       ? null
```
Comment (Mic92) [inline pkgs/applications/video/mpv/default.nix]: Why is this removed?

Comment (rnhmjoj) [reply pkgs/applications/video/mpv/default.nix]: I moved these to the package call (in all-packages.nix) since there was a `waylandSupport = stdenv.isLinux;` there already.

**Comment by Mic92** [comment]
Mhm which package introduce `libpciaccess` in mpv?

**Comment by rnhmjoj** [comment]
I think we need a `drmSupport = !stdenv.isDarwin`.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -16541,11 +16541,16 @@ with pkgs;
...
-    libva = libva-full;
-    waylandSupport = stdenv.isLinux;
+    libva      = libva-full;
+    waylandSupport     = stdenv.isLinux;
+    alsaSupport        = !stdenv.isDarwin;
+    pulseSupport       = !stdenv.isDarwin;
+    rubberbandSupport  = !stdenv.isDarwin;
+    dvdreadSupport     = !stdenv.isDarwin;
+    dvdnavSupport      = !stdenv.isDarwin;
+    drmSupport         = !stdenv.isDarwin;
```
Comment (Mic92) [inline pkgs/top-level/all-packages.nix]: drmSupport is already disabled.

Comment (rnhmjoj) [reply pkgs/top-level/all-packages.nix]: Oh, right. So does this mean the `? true` in the package is overriding the option in the call?

Comment (Mic92) [reply pkgs/top-level/all-packages.nix]: Only if the variable is null, which is not the case.
Does `why-depends` only show the first match?

```console
$ nix-repl> ({ a ? true }: a) { a = false; }
false
```

File: pkgs/applications/video/mpv/default.nix [hunk 3]
```diff
@@ -1,7 +1,7 @@
 { stdenv, fetchurl, fetchFromGitHub, fetchpatch, makeWrapper
 , docutils, perl, pkgconfig, python3, which, ffmpeg
 , freefont_ttf, freetype, libass, libpthreadstubs
-, lua, lua5_sockets, libuchardet, libiconv ? null, darwin
+, lua, luasocket, libuchardet, libiconv ? null, darwin
 
 , x11Support ? true,
```
Comment (Mic92) [inline pkgs/applications/video/mpv/default.nix]: `libpciaccess` is pulled by mesa, which is pulled by x11Support.

**Comment by Mic92** [comment]
@GrahamcOfBorg build mpv

**Comment by Mic92** [comment]
Let's roll the dice.

**Comment by Mic92** [comment]
@GrahamcOfBorg build mpv

File: pkgs/top-level/all-packages.nix (hunk 2 above)
Comment (Mic92) [reply pkgs/top-level/all-packages.nix]:
```console
$ nix why-depends -a nixpkgs.mpv nixpkgs.xlibs.libpciaccess
```

gives a more complete picture

**Comment by rnhmjoj** [comment]
Thank you
